scripts/helpers

```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/env bash
+
+set -Eeuo pipefail
+
+# Validates dev_tags for DockerHub only pushes to "hashicorppreview" org
+dev_tags_validation() {
+  if [ -n "${1}" ]; then
+    for dt in ${1}; do
+      # strip docker.io if exists
+      dt_stripped=${dt//docker.io\//}
+
+      # if there are two parts we presume it's the public docker registry, ex: hashicorppreview/consul. Anything else is a registry we don't check currently.
+      parts=$(echo "$dt_stripped" | awk -F'/' '{print NF}')
+      if [ "$parts" -eq 2 ] && [[ "$dt_stripped" != hashicorppreview/* ]]; then
+        echo "Only valid DockerHub organization is 'hashicorppreview' (Got: $dt)"
+        exit 1
+      fi
+    done
+  fi
```

Comment:
I wonder if we should handle the case where a user tries to specify an ECR or Red Hat tag in `dev_tags` as well?

I think of this policy as more of a whitelist than a blacklist. So, any tag which doesn't match `^(docker\.io/)?hashicorppreview/.*` should be rejected I think? Would you be ok to try that implementation @alvin-huang ? I hope it can be done with a single grep that way so should be a bit simpler.